Cabal/Distribution/Simple/GHC.hs

```diff
@@ -561,10 +561,13 @@ buildOrReplLib forRepl verbosity numJobs pkg_descr lbi lib clbi = do
                                }
                odir          = fromFlag (ghcOptObjDir vanillaCcOpts)
            createDirectoryIfMissingVerbose verbosity True odir
-           runGhcProg vanillaCcOpts
-           unless forRepl $
-             whenSharedLib forceSharedLib (runGhcProg sharedCcOpts)
-           unless forRepl $ whenProfLib (runGhcProg profCcOpts)
+           let oname = odir </> dropExtension filename <.> fromFlagOrDefault "o" (ghcOptObjSuffix vanillaCcOpts)
```

Comment:
This looks like it should be made a helper function. Also not sure if `<.>` is available in all versions of `filepath` that we support.